testing/test_compile_wasm_to_memory.sc

```diff
@@ -0,0 +1,15 @@
+fn sum (x y)
+    x + y
+
+let result = 
+    compile-wasm-to-buffer
+        "test"
+        compiler-file-kind-object
+        do
+            let 
+                func =
+                    static-typify sum i32 i32
+            locals;
+        #
+
+print result
```

Comment:
Ideally we'd have a test here that compares the memory buffer output to writing it out to a file and makes sure this is identical to the existing WASM tests, but that's not mandatory

Reply:
I'm not actually certain if they're the same, this vomits out a human readable ASM whereas the other WASM file is a literal binary. Some further LLVM fiddling may need to be done, @aiverson will let me know I'm sure.